src/evaluatorfrontend.cpp

```diff
@@ -67,18 +74,82 @@ void EvaluatorFrontend::FuncRunner() {
 		Plot();
 }
 
-//! @file
-// [runeval]
-void EvaluatorFrontend::RunEvaluator() {
-	while (!evaluator->IsFinished()) {
+int EvaluatorFrontend::EvaluatorFunc() {
+	if (!evaluator->IsFinished()) {
 		states.push_back(evaluator->GetNextNetworkState());
 		if (printStd) {
 			std::cout << states.back().PrintCsvRow();
 		}
+	} else {
+		return 0;
 	}
+
+	return 1;
+}
+
+//! @file
+// [runeval]
+void EvaluatorFrontend::RunEvaluator() {
+	while (EvaluatorFunc())
+		;
 }
 // [runeval]
 
+void EvaluatorFrontend::RunRTEvaluator() {
+	auto StartT = std::chrono::steady_clock::now();
+	auto EndT = std::chrono::steady_clock::now();
+	bool once = true;
+
+	while (!mPlot->RunPlot()) {
```

Comment:
Looping forever after we're done might not be the best way to do it. Maybe instead do something like printing `Simulation complete. Press Enter to exit application`, and then waiting for keyboard input is a viable alternative. ALthough, this can wait till the next PR when we implement multithreading.

Reply:
The RunPlot function is actually the update loop for the main window and opengl, so the loop will run until either opengl crashes or the user tells it to shut down

Reply:
Yeah, but the program doesn't hand the CPU back to the kernel. I can't test it, due to being on a low glfw version, but I'd suspect that the non-running program would have a pretty high CPU usage

Reply:
It's programmed as a "standard" game loop, since CRNSimul is responsible for OpenRTPlotter, it should also be responsible for the window (they share the same thread).

This is also the "same" way the glfw examples does it.

Also it stops looping when the window shuts down (So it's not forever)

Reply:
Okay, so I looked it up, and from what I can gather, from a very quick look, glfw will limit framerate to 60 fps on its own, making any sort of handing CPU cycles on our part unnecessary.